packages/calypso-razorpay/src/index.tsx

```diff
@@ -0,0 +1,249 @@
+	const memoizedRequestArgs = useMemoCompare( requestArgs, areRequestArgsEqual );
+
+	useEffect( () => {
+		debug( 'Loading razorpay configuration' );
+		let isSubscribed = true;
+		fetchRazorpayConfiguration( memoizedRequestArgs || { sandbox: true } )
+			.then( ( configuration ) => {
+				if ( ! isSubscribed ) {
+					return;
+				}
+				if ( ! configuration.js_url ) {
+					debug( 'Invalid razorpay configuration; js_url missing' );
+					debug( configuration );
+					throw new RazorpayConfigurationError(
+						'Error loading payment method configuration. Received invalid data from the server.'
+					);
+				}
+				debug( 'Razorpay configuration received' );
+				debug( configuration );
+				setRazorpayConfiguration( configuration ?? null );
+			} )
+			.catch( ( error ) => {
+				setRazorpayConfigurationError( error );
+			} );
+		return () => {
+			isSubscribed = false;
+		};
+	}, [ memoizedRequestArgs, fetchRazorpayConfiguration ] );
+	return { razorpayConfiguration, razorpayConfigurationError };
+}
+
+// TODO
+function areRequestArgsEqual(
+	previous: undefined | null | GetRazorpayConfigurationArgs,
+	next: undefined | null | GetRazorpayConfigurationArgs
+): boolean {
+	return !! previous && !! next; // TODO
+}
+
+export function RazorpayHookProvider( {
+	children,
+	fetchRazorpayConfiguration,
+}: PropsWithChildren< {
+	fetchRazorpayConfiguration: GetRazorpayConfiguration;
+} > ) {
+	const configurationArgs = {
+		sandbox: true,
+	};
+
+	const { razorpayConfiguration, razorpayConfigurationError } = useRazorpayConfiguration(
+		fetchRazorpayConfiguration,
+		configurationArgs
+	);
+
+	const { razorpay, isRazorpayLoading, razorpayLoadingError } = useRazorpayJs(
+		razorpayConfiguration,
+		razorpayConfigurationError
+	);
+
+	const razorpayData = {
+		razorpay,
+		razorpayConfiguration,
+		isRazorpayLoading,
+		razorpayLoadingError,
+	};
+
+	return <RazorpayContext.Provider value={ razorpayData }>{ children }</RazorpayContext.Provider>;
+}
+
+/**
+ * Custom hook to access Razorpay.js
+ *
+ * First you must wrap a parent component in `RazorpayHookProvider`. Then you can
+ * call this hook in any sub-component to get access to the razorpay variables
+ * and functions.
+ *
+ * This returns an object with the following properties:
+ *
+ * - razorpay: the instance of the razorpay library
+ * - razorpayConfiguration: the object containing the data returned by the wpcom razorpay configuration endpoint
+ * - isRazorpayLoading: a boolean that is true if razorpay is currently being loaded
+ * - razorpayLoadingError: an optional object that will be set if there is an error loading razorpay
+ * @returns {RazorpayData} See above
+ */
+export function useRazorpay(): RazorpayData {
+	const razorpayData = useContext( RazorpayContext );
+	if ( ! razorpayData ) {
+		throw new Error( 'useRazorpay can only be used inside a RazorpayHookProvider' );
+	}
+	return razorpayData;
+}
+
+// See https://usehooks.com/useMemoCompare/
```

Comment:
That URL doesn't seem to work anymore and I don't see see it in the usehooks directory either 🤔

Reply:
Interesting. I guess they must have removed it? Well, we've been using this pattern successfully for a while. We might just want to remove the comment.